src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,101 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.parser.Parser;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/** Command is the base class of all commands */
+public abstract class Command {
+    private final CommandType commandType;
+    private final String commandString;
+
+    /**
+     * Constructor for command class
+     *
+     * @param commandType Type of command
+     * @param commandString String of actual command
+     */
+    public Command(CommandType commandType, String commandString) {
+        this.commandType = commandType;
+        this.commandString = commandString;
+    }
+
+    /**
+     * Returns a Command subclass object based on the commandType and commandString given
+     *
+     * @param commandType of the commandString
+     * @param commandString represent the command
+     * @return a Command subclass object based on the commandType and commandString given
+     * @throws DukeException when the command is invalid
+     */
+    public static Command create(CommandType commandType, String commandString) throws DukeException {
+        switch (commandType) {
```

Comment:
Good to see your indent for switch statements as well as adding a default case!

Reply:
Thanks!